### PR TITLE
Feature/558 text ts changes

### PR DESCRIPTION
**Pull request by rma-rripken**

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/RegularTextTimeSeriesRow.java [hunk 1]
```diff
@@ -51,26 +39,23 @@ public String getTextValue() {
...
+
+    public Instant getDataEntryDate() {
         return dataEntryDate;
     }
 
     @JsonIgnore
-    @Override
     public DateDateKey getDateDateKey() {
-        return new DateDateKey(dateTime, dataEntryDate);
+        return new DateDateKey(dateTime == null? null: Date.from(dateTime), dataEntryDate==null ? null: Date.from(dataEntryDate));
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/RegularTextTimeSeriesRow.java]: would it be better to have a second constructor that handles this? I think I saw this exact logic in another spot.

Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/RegularTextTimeSeriesRow.java]: Possibly.  BinaryTimeSeriesRow is probably the other location. That class also transitioned from Date to Instant but kept DateDateKey.  Will revisit this before taking the pr out of draft.  Maybe DateDateKey goes away entirely.

**Review by MikeNeilson — COMMENTED** [review]
Marking place for later review.

**Review by MikeNeilson — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/RegularTextTimeSeriesRow.java (hunk 1 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/RegularTextTimeSeriesRow.java]: Turns out this is using DateDateKey from Monolith.  Adding a new constructor isn't worth the effort (in my opinion) so I added a build method.   I don't want to do it right now but I could a follow-up pr for deleting DateDateKey from the text-ts and bin-ts.

**Review by MikeNeilson — DISMISSED** [review]
That looks good.

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**